### PR TITLE
hyperscan: better error message if not compiled

### PR DESCRIPTION
**Pull request by inashivb**

### CODE REVIEW

File: src/detect-engine-mpm.c [hunk 1]
```diff
@@ -643,6 +643,12 @@ uint16_t PatternMatchDefaultMatcher(void)
             }
         }
 
+#ifndef BUILD_HYPERSCAN
+    if (strcmp(mpm_algo, "hs") == 0) {
+        FatalError(SC_ERR_INVALID_VALUE, "Hyperscan (hs) support is "
```
Comment (victorjulien) [inline src/detect-engine-mpm.c:648]: this error provides no guidance or even enough context for users with little experience. Would suggest adding that it is about the mpm-algo option value.

File: src/util-spm.c [hunk 2]
```diff
@@ -83,6 +83,12 @@ uint16_t SinglePatternMatchDefaultMatcher(void)
             }
         }
 
+#ifndef BUILD_HYPERSCAN
+        if (strcmp(spm_algo, "hs") == 0) {
+            FatalError(SC_ERR_INVALID_VALUE, "Hyperscan (hs) support is "
```
Comment (victorjulien) [inline src/util-spm.c:88]: same but for spm-algo

File: src/detect-engine-mpm.c (hunk 1 above)
Comment (inashivb) [reply src/detect-engine-mpm.c:648]: > this error provides no guidance or even enough context for users with little experience. Would suggest adding that it is about the mpm-algo option value.

That's correct. I ran it in dev mode and could see the file names so missed it.

How is `"Hyperscan (hs) support for mpm-algo is not compiled into Suricata"`?

Comment (jlucovsky) [reply src/detect-engine-mpm.c:648]: Add "Add`-with-libhs-includes and -with-libhs-libraries to configure."?

**Comment by inashivb** [comment]
Replaced with #5149

**Pull request closed without merging**